### DISCUSSION TITLE
Dynamiclly adding new pointcloud to an presented scenegraph

### DISCUSSION THREAD

**Discussion by HBK1JD** [vsg-dev/VulkanSceneGraph#998] [Q&A]
Hi Robert,
I met a problem while building my vsgQt project. I am trying to add a new point cloud to a scene graph which is already presented in a viewer. I have successfully updated the point cloud as a node using `scene->addChild()` to my scene, but after I tried

`auto result = viewer->compileManager->compile(node);`
`vsg::updateViewer(*viewer, result);`

to update the viewer, the new point cloud failed to presented and my window freezed. Did I miss something  or do something wrong? Thanks for any advice!

**Comment by robertosfield** [comment]
I know too little about what exactly you are doing when and what the error might be to speculate what might be the problem.

The best I can do is suggest steps for you to take to get more information about what is going wrong, and to provide some general tips.

1. First up run you application with the Vulkan debug layer enabled to see what errors are being produced.  You'll need to se the  WindowTraits::debugLayer variable to true, see the vsgqtviewer etc. examples for this being done.  Then look at the output. The apiDumpLayer can also be useful to set as a second pass to help figure out where any error is occurring.
2. When the application freezes break it in a debugger and have a look at the stack trace.
3. If you are doing things multi-thread you'll need to the scene graph and compile in background thread, but merge in main viewer thread, use as via an update operation.  See the vsgdynamicload example.
